contributor-docs/code-change-guide.md

```diff
@@ -423,6 +423,17 @@ If you're using Dataflow Runner v2 and `sdks/java/harness` or its dependencies (
   --sdkContainerImage="us.gcr.io/apache-beam-testing/beam_java11_sdk:2.49.0-custom"
   ```
 
+#### Snapshot Version Containers
+
+By default, a Snapshot version for an SDK under development will use the containers published to the [apache-beam-testing project's container registry](https://us.gcr.io/apache-beam-testing/github-actions).For example, the most recent snapshot container for Java 17 can be found [here](https://us.gcr.io/apache-beam-testing/github-actions/beam_java17_sdk).
```

Comment:
The "2.xx.0-SNAPSHOT" downtime is longer for snapshot jars (publish nightly) than snapshot containers (publish every 4 hrs). And I read that the snapshot container should not affect if user follows this doc, but snapshot jar does.

Shall we change the section title to "Snapshot Artifacts and Version Containers" ?

Reply:
I'm not sure I 100% understand the concern here. I think Jars not being available is fine since we'll continue to use the old snapshot jars until the new ones are published.

> if sdkContainer not provided, it goes to an old beam master container; otherwise it goes to the override one

This is true until we get to the final snapshot which is published as part of the release. After cutting the release branch, we:

1) Update the containers used - https://github.com/apache/beam/commit/84daae3610bbe3037cb6fbd9d286c4bbb386990e - these containers haven't been published yet
2) Publish a snaphsot off of the release branch - https://github.com/apache/beam/blob/master/contributor-docs/release-guide.md#cut-the-release-branch - this uses the 2.xx.0 containers, not the 2.xx.0-SNAPSHOT containers. These haven't been published yet
3) Publish those containers

During the gap between (2) and (3), snapshot versions won't be able to find containers (which I think is ok)

Reply:
I understand now, yeah the problem is step 1 changed to beam-master to release versioned container